structure/bitmap_1.go

```diff
@@ -0,0 +1,476 @@
+package structure
+
+import (
+	"encoding/binary"
+	"errors"
+	"sort"
+
+	"github.com/ByteStorage/FlyDB/config"
+	"github.com/ByteStorage/FlyDB/engine"
+	_const "github.com/ByteStorage/FlyDB/lib/const"
+)
+
+type BitmapStructure struct {
+	db *engine.DB
+}
+
+// NewBitmapStructure returns a new BitmapStructure
+// It will return a nil BitmapStructure if the database cannot be opened
+// or the database cannot be created
+// The database will be created if it does not exist
+func NewBitmapStructure(options config.Options) (*BitmapStructure, error) {
+	db, err := engine.NewDB(options)
+	if err != nil {
+		return nil, err
+	}
+	return &BitmapStructure{db: db}, nil
+}
+
+// SetBit Set the bit at the specified offset in the bitmap to the specified value
+// If the key does not exist, it will be created
+// If the bitmap length is not sufficient, it will be extended
+func (b *BitmapStructure) SetBit(key []byte, offset uint, value bool) error {
```

Comment:
Function input is `string` type, please note the modification.

The same below